src/openvswitch.c

```diff
@@ -29,6 +29,42 @@
 #include "util.h"
 #include "util-internal.h"
 
+#ifndef PREFIX 
+#define PREFIX "/usr/"
+#endif
+
+#ifndef BINDIR
+#define BINDIR PREFIX"bin"
+#endif
+
+#ifndef SNAPBINDIR
+#define SNAPBINDIR "/snap/bin"
+#endif
+
+#define OPENVSWITCH_OVS_VSCTL "ovs-vsctl"
+
+static GString *netplan_openvswitch_ovs_vsctl_path = NULL;
+
+void netplan_openvswitch_vsctl__(void)
+{
+    netplan_openvswitch_ovs_vsctl_path = g_string_new(NULL);
+
+    if (g_file_test(SNAPBINDIR "/" OPENVSWITCH_OVS_VSCTL, G_FILE_TEST_EXISTS)) {
+        g_string_printf(netplan_openvswitch_ovs_vsctl_path, "%s/%s", SNAPBINDIR, OPENVSWITCH_OVS_VSCTL);
+    } else {
+        g_string_printf(netplan_openvswitch_ovs_vsctl_path, "%s/%s", BINDIR, OPENVSWITCH_OVS_VSCTL);
+    }
+}
+
+const char *netplan_openvswitch_ovs_vsctl_(void)
```

Comment:
note: This is a `visibility: hidden` function by default. So the naming scheme doesn't matter too much. By convention `netplan_*` functions are mostly used for public API, though. Maybe we should rather call it `_get_netplan_openvswitch_ovs_vsctl_path`?